.travis.yml

```diff
@@ -13,8 +13,3 @@ script:
   - ./.ci/travis.run
 after_failure:
   - cat build/test-suite.log
-after_success:
```

Comment:
Is this really the problem? If I can access build/test-suite.log then why can't I access the other data? I share this folder with the docker container so that when it's done, I can get any data I need.

If this is the problem, then why do I have any coverage data uploaded?

Reply:
I'm not 100% sure - but by putting it in the docker file, as indicated in the documentation it works.

The only test coverage you get is by twist which is a unit test (compared to integration) - which is indeed strange.

By examining the output of the two solutions, 
travis.yml version outputs
```
==> Travis CI detected.
    project root: .
--> token set from env
    Yaml found at: codecov.yml
==> Running gcov in . (disable via -X gcov)
```

docker version outputs
```
==> Travis CI detected.
    project root: /workspace/tpm2-pkcs11
    Yaml found at: codecov.yml
==> Running gcov in /workspace/tpm2-pkcs11 (disable via -X gcov)
```

Maybe it the path needs to be specified absolutely or something similar.

The folder is shared, as in the documentation, but in a little bit different manner
`docker run -v "$PWD/shared:/shared" ...`
vs
`docker run -v `pwd`:/workspace/tpm2-pkcs11`

So the directories are mapped a little bit differently - maybe that's the reason.

Reply:
The other problem, when running codecov from travis and not from docker, is that the version of gcc and gcov are mismatching.
This is where this version **'504*', prefer '408*'** comes from.

The docker environment uses "gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609"
The travis environment used gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4

Maybe we also should consider updating our travis env (and docker) to xenial. (18.04)
Trusty is 5 years old now....

Reply:
Yep agree. I'd like us to have multiple images, and cron-job master on a few different distros and versions and then check PR's against something like the newest stable Ubuntu.